rcl_yaml_param_parser/test/test_parse_yaml.cpp

```diff
@@ -23,19 +23,57 @@
 
 static char cur_dir[1024];
 
-TEST(test_file_parser, correct_syntax) {
+TEST(test_parser, correct_syntax) {
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024));
+  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   char * path = rcutils_join_path(test_path, "correct_config.yaml", allocator);
-  fprintf(stderr, "cur_path: %s\n", path);
-  EXPECT_TRUE(rcutils_exists(path));
+  ASSERT_TRUE(rcutils_exists(path)) << "No test YAML file found at " << path;
```

Comment:
If we use `ASSERT_*`, I think we should make sure to free resources in the event that the assertion fails. We can use the helper [`OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT`](https://github.com/osrf/osrf_testing_tools_cpp/blob/fde512a1c9ff059cd151e366116baad05435e01f/osrf_testing_tools_cpp/include/osrf_testing_tools_cpp/scope_exit.hpp#L45) to make sure `test_path`, `path`, and `params_hdl` are cleaned up when the test exits.

I guess this should also be added to the other tests as needed.

Reply:
I thought about it, but since it's a test I was less worried about resource freeing. But using `OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT` sounds like a much better plan.

Reply:
Previously, some folks went through and fixed a bunch of leaks detected by ASan jobs ([for example](https://github.com/ros2/rcl/pull/442)). Even though they're just tests, it will make those jobs happy.